src/builder.rs

```diff
@@ -0,0 +1,98 @@
+use std::sync::Arc;
+
+use anyhow::{anyhow, Result};
+use hyper::{Body, Request, Response};
+
+use crate::app_context::AppContext;
+use crate::async_graphql_hyper::{GraphQLBatchRequest, GraphQLRequest};
+use crate::blueprint::{validate_operations, Blueprint, OperationQuery};
+use crate::config::reader::ConfigReader;
+use crate::config::{Config, ConfigModule, Source};
+use crate::fmt::Fmt;
+use crate::http::handle_request;
+use crate::print_schema;
+use crate::runtime::TargetRuntime;
+use crate::valid::{ValidationError, Validator};
+
+#[derive(Clone)]
+pub struct TailcallBuilder {
+    runtime: TargetRuntime,
+}
+#[derive(Clone)]
+pub struct TailcallExecutor {
+    pub config_module: ConfigModule,
+    pub app_ctx: Arc<AppContext>,
+}
+
+impl TailcallBuilder {
+    pub fn init(runtime: TargetRuntime) -> Self {
+        Self { runtime }
+    }
+    pub async fn with_config<T: ToString>(
+        self,
+        source: Source,
+        schema: T,
+        relative_path: Option<String>,
+    ) -> Result<TailcallExecutor> {
+        let reader = ConfigReader::init(self.runtime.clone());
+        let config = Config::from_source(source, &schema.to_string())?;
+
+        let config_module = reader
+            .resolve(config, relative_path)
+            .await
+            .map_err(|e| ValidationError::new(e.to_string()))?;
+        let blueprint = Blueprint::try_from(&config_module.clone())?;
+        let app_ctx = AppContext::new(blueprint, self.runtime);
+        let app_ctx = Arc::new(app_ctx);
+        Ok(TailcallExecutor { config_module, app_ctx })
+    }
+    pub async fn with_config_paths<T: ToString>(self, files: &[T]) -> Result<TailcallExecutor> {
+        let reader = ConfigReader::init(self.runtime.clone());
+        let config_module = reader.read_all(files).await?;
+        let blueprint = Blueprint::try_from(&config_module)?;
+        let app_ctx = AppContext::new(blueprint, self.runtime);
+        let app_ctx = Arc::new(app_ctx);
+        Ok(TailcallExecutor { config_module, app_ctx })
+    }
+}
+
+impl TailcallExecutor {
+    pub async fn validate(
+        &self,
+        n_plus_one_queries: bool,
+        schema: bool,
+        ops: Vec<OperationQuery>,
```

Comment:
Vec<String> and if decode to Operation Query fails It should give ValidationError

Reply:
add a todo for now